File: Perl/closure/1/main.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub main {
    say(sub {
        my $content = shift;
        print $content;
        return "Hello Perl!\n";
    });
}

sub say {
    my $reply = shift;
    print $reply->("Hello World!\n");
}

main();

File: JavaScript/closure/1/main.js
/**
 * Created by fay on 2017/6/1.
 */

function main() {
    say(function (content) {
        console.log(content);
        return "Hello JavaScript!";
    })
}

function say(reply) {
    console.log(reply("Hello World!"));
}

main();
